Split/Engine/DefaultTreatmentManager.swift

```diff
@@ -0,0 +1,158 @@
+        return result
+    }
+    
+    func getTreatment(_ splitName: String, attributes: [String : Any]?) -> String {
+        let timeMetricStart = Date().unixTimestampInMicroseconds()
+        let result = getTreatmentWithConfigNoMetrics(splitName: splitName, shouldValidate: true, attributes: attributes, validationTag: ValidationTag.getTreatment).treatment
+        metricsManager.time(microseconds: Date().unixTimestampInMicroseconds() - timeMetricStart, for: Metrics.time.getTreatment)
+        return result
+    }
+    
+    func getTreatments(splits: [String], attributes:[String:Any]?) ->  [String:String] {
+        let timeMetricStart = Date().unixTimestampInMicroseconds()
+        let result = getTreatmentsWithConfigNoMetrics(splits: splits, attributes: attributes, validationTag: ValidationTag.getTreatments).mapValues { $0.treatment }
+        metricsManager.time(microseconds: Date().unixTimestampInMicroseconds() - timeMetricStart, for: Metrics.time.getTreatments)
+        return result
+    }
+    
+    func getTreatmentsWithConfig(splits: [String], attributes:[String:Any]?) ->  [String:SplitResult] {
+        let timeMetricStart = Date().unixTimestampInMicroseconds()
+        let result = getTreatmentsWithConfigNoMetrics(splits: splits, attributes: attributes, validationTag: ValidationTag.getTreatmentsWithConfig)
+        metricsManager.time(microseconds: Date().unixTimestampInMicroseconds() - timeMetricStart, for: Metrics.time.getTreatmentsWithConfig)
+        return result
+    }
+    
+    private func getTreatmentsWithConfigNoMetrics(splits: [String], attributes:[String:Any]?, validationTag: String) ->  [String:SplitResult] {
+        var results = [String:SplitResult]()
+        
+        if let errorInfo = keyValidator.validate(matchingKey: key.matchingKey, bucketingKey: key.bucketingKey) {
+            validationLogger.log(errorInfo: errorInfo, tag: validationTag)
+            return splits.filter { !$0.isEmpty() }.reduce([String: SplitResult]()) { results, splitName in
+                var res = results
+                res[splitName] = SplitResult(treatment: SplitConstants.CONTROL)
+                return res
+            }
+        }
+        
+        if splits.count > 0 {
+            let splitsNoDuplicated = Set(splits.filter { !$0.isEmpty() }.map { $0 })
+            for splitName in splitsNoDuplicated {
+                results[splitName] = getTreatmentWithConfigNoMetrics(splitName: splitName, shouldValidate: false, attributes: attributes, validationTag: validationTag)
+            }
+        } else {
+            Logger.d("\(validationTag): split_names is an empty array or has null values")
+        }
+        return results
+    }
+    
+    private func getTreatmentWithConfigNoMetrics(splitName: String, shouldValidate: Bool = true, attributes:[String:Any]? = nil, validationTag: String) -> SplitResult {
+        
+        if shouldValidate {
```

Comment:
Yup. That flag is used to avoid validation for each split when is called from getTreatments. In this cases the validation occurs in the calling function.